pkg/rbac/validate_test.go

```diff
@@ -287,3 +289,21 @@ func TestCan(t *testing.T) {
 		})
 	}
 }
+
+func TestObjectMatch(t *testing.T) {
+	t.Parallel()
+	assert.True(t, objectMatch("", "*"))
+	assert.True(t, objectMatch("namespace-1/object-1", "*"))
+	assert.True(t, objectMatch("/", "*/*"))
+	assert.True(t, objectMatch("namespace-1/", "namespace-1/*"))
+	assert.True(t, objectMatch("namespace-1/object-1", "namespace-1/*"))
+	assert.False(t, objectMatch("namespace-2/", "namespace-1/*"))
+	assert.False(t, objectMatch("namespace-2/object-1", "namespace-1/*"))
+	assert.True(t, objectMatch("namespace-1/", "*/*"))
+	assert.True(t, objectMatch("namespace-1/object-1", "*/*"))
+	assert.False(t, objectMatch("namespace-1/object-1", "namespace-2/object-1"))
+	assert.False(t, objectMatch("namespace-2/object-2", "namespace-2/object-1"))
+	assert.False(t, objectMatch("namespace-2/object-2", "*/object-1"))
+	assert.True(t, objectMatch("namespace-2/object-1", "*/object-1"))
```

Comment:
This is the use-case we needed, which was not covered originally by the `keyMatch` function (but covered in globMatch)